sdk/core/azure-core/test/ut/log_policy_test.cpp

```diff
@@ -61,7 +61,33 @@ void SendRequest(LogOptions const& logOptions, std::string const& portAndPath =
 
   request.SetHeader("hEaDeR1", "HvAlUe1");
   request.SetHeader("HeAdEr2", "hVaLuE2");
-  request.SetHeader("x-ms-request-id", "6c536700-4c36-4e22-9161-76e7b3bf8269");
+
+  // Add in all the default allowed HTTP headers to the request. We'll make sure they're not
+  // redacted on the way out.
+  request.SetHeader("Accept", "Accept");
+  request.SetHeader("Cache-Control", "Cache-Control");
+  request.SetHeader("Connection", "Connection");
+  request.SetHeader("Content-Length", "Content-Length");
+  request.SetHeader("Content-Type", "Content-Type");
+  request.SetHeader("Date", "Date");
+  request.SetHeader("ETag", "ETag");
+  request.SetHeader("Expires", "Expires");
+  request.SetHeader("If-Match", "If-Match");
+  request.SetHeader("If-Modified-Since", "If-Modified-Since");
+  request.SetHeader("If-None-Match", "If-None-Match");
+  request.SetHeader("If-Unmodified-Since", "If-Unmodified-Since");
+  request.SetHeader("Last-Modified", "Last-Modified");
+  request.SetHeader("Pragma", "Pragma");
+  request.SetHeader("Request-Id", "Request-Id");
+  request.SetHeader("Retry-After", "Retry-After");
+  request.SetHeader("Server", "Server");
+  request.SetHeader("traceparent", "traceparent");
+  request.SetHeader("tracestate", "tracestate");
+  request.SetHeader("Transfer-Encoding", "Transfer-Encoding");
+  request.SetHeader("User-Agent", "User-Agent");
+  request.SetHeader("x-ms-client-request-id", "x-ms-client-request-id");
+  request.SetHeader("x-ms-request-id", "x-ms-request-id");
+  request.SetHeader("x-ms-return-client-request-id", "x-ms-return-client-request-id");
```

Comment:
This clutters the test case a lot. Please have a boolean constructor parameter whether to add all headers, false by default, and only add all these headers if that parameter is true. Then add one test case that tests the default headers and nothing else. Leave the existing test cases without changing.